app/jobs/autoscaling.py

```diff
@@ -260,9 +440,14 @@ def autoscaling_update(my_num_jobs, my_environment="development"):
         # when debugging is disabled and we can't scale we exit early here
         if not DEBUG:
             return
+    tag_list = get_instance_tags()
+    gsnap_tags = find_tags_for_instances_in(gsnap_asg, tag_list)
+    rapsearch_tags = find_tags_for_instances_in(rapsearch2_asg, tag_list)
+    check_draining_servers(gsnap_asg, gsnap_tags, draining_tag, job_tag_prefix, job_tag_expiration, min_draining_wait, can_scale)
```

Comment:
Is there a reason check_draining_servers runs before set_desired_capacity?   I would have thought in the interest of reclaiming optimization you would wish to set desired capacity first, to give yourself the ability to reclaim some draining servers, instead of removing their protection at the start which makes them unavailable for reclaiming later.

Reply:
Good point. I will move it to the end